packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,687 @@
+## Introduction
+
+This tutorial is intended for advanced users who are familiar with basic [Grid's](./getting-started.md) concept and who wants to learn how to use [React Core](../../../core/docs/guides/fundamentals.md) components for extending the Grid component functionality.
+
+In our particular case, we will add a select box in the Grid's toolbar that will allow users to filter the grid by the required column. The purpose of the tutorial is not to implement a very useful feature. The main idea is to show how to add functionality to the Grid componet that does not exist out of the box.
+
+We chose the `Material-UI` theme and our initial code is follows:
+
+*index.js:*
+```jsx
+import {
+  Grid,
+  Table,
+  TableHeaderRow,
+  Toolbar
+} from '@devexpress/dx-react-grid-material-ui';
+import service from './data.js';
+
+class App extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        { name: "ID", title: "ID" },
+        { name: "CompanyName", title: "Company Name" },
+        { name: "Address", title: "Address" },
+        { name: "City", title: "City" },
+        { name: "State", title: "State" },
+        { name: "Zipcode", title: "Zip code" }
+      ],
+      rows: service.getCustomers()
+    };
+  }
+  render() {
+    const { rows, columns } = this.state;
+    return (
+      <Paper>
+        <Grid
+          rows={rows}
+          columns={columns}
+        >
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+*data.js:*
+```js
+const customers = [{
+    ID: 1,
+    CompanyName: 'Super Mart of the West',
+    Address: '702 SW 8th Street',
+    City: 'Bentonville',
+    State: 'Arkansas',
+    Zipcode: 72716
+  }, {
+    ID: 2,
+    CompanyName: 'Electronics Depot',
+    Address: '2455 Paces Ferry Road NW',
+    City: 'Atlanta',
+    State: 'Georgia',
+    Zipcode: 30339
+  }, {
+    ID: 3,
+    CompanyName: 'K&S Music',
+    Address: '1000 Nicllet Mall',
+    City: 'Minneapolis',
+    State: 'Minnesota',
+    Zipcode: 55403
+  }, {
+    ID: 4,
+    CompanyName: "Tom's Club",
+    Address: '999 Lake Drive',
+    City: 'Issaquah',
+    State: 'Washington',
+    Zipcode: 98027
+  }, {
+    ID: 5,
+    CompanyName: 'E-Mart',
+    Address: '3333 Beverly Rd',
+    City: 'Hoffman Estates',
+    State: 'Illinois',
+    Zipcode: 60179
+  }];
+
+export default {
+  getCustomers() {
+    return customers;
+  }
+}
+```
+
+Let's take a look at the screenshot below to see the final result that we will get after performing all the steps from the tutorial:
+
+![](../../../../../img/guides/custom-plugin-development/result.png)
```

Comment:
I suggest using a `html` for pass images because we should add specials styles for good show these for other devices.

I suggest adding a class 'img-responsive' for resolve problems described above.

```jsx
<p align="center">
  <img class="img-responsive" src="img-path.img.png">
</p>
```